Добавил автотесты

SergeiShein opened the pull request:
Параметризацию не понял где именно нужно применить, сделал без нее

SergeiShein committed:
Добавил автотесты
Параметризацию не понял где именно нужно применить, сделал без нее

alesyaisachenko commented on src/main/java/com/example/Lion.java:
```diff
@@ -16,7 +16,10 @@ public Lion(String sex) throws Exception {
 
-    Feline feline = new Feline();
+    Feline feline;
+    public Lion (Feline feline){
```
Это нужно исправить.


изоляцию Lion от Feline нужно было делать через инъекцию зависимостей. То есть в класс Lion должен попадать уже созданный экземпляр класса Feline.

В классе должен быть только  один конструктор public Lion(String sex, Feline feline).

alesyaisachenko commented on src/test/java/AutoTest.java:
```diff
@@ -0,0 +1,82 @@
+import java.util.List;
+
+@RunWith(MockitoJUnitRunner.class)
+public class AutoTest {
```
Это нужно исправить:

Этот тестовый класс нужно разбить на несколько тестовых  классов. Для каждого объекта (Lion, Feline, Cat) нужно создать отдельный тестовый класс.

alesyaisachenko commented on src/test/java/AutoTest.java:
```diff
@@ -0,0 +1,82 @@
+        List<String> meat = feline.eatMeat();
+        assertEquals(List.of("Животные", "Птицы", "Рыба"), meat);
+        String family = feline.getFamily();
+        assertEquals("Кошачьи", family);
```
Это нужно исправить:

для каждого метода нужно создать свой отдельный тест, а не тестировать все методы в одном тесте. Причина проста -- если упадет тест на первой проверке, то остальные методы не будут проверяться.

Подправь, плиз, этот момент во всем коде.

SergeiShein committed:
Исправил замечания.
Я не понимаю, где добавить параметризацию, уже написал это в прошлом коммите, если подскажите будет очень классно.

SergeiShein committed:
Добавил параметризацию.
